File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.js
function solution(targets) {
    
    targets.sort((a,b) => {
        if(a[1] == b[1]) return a[0] - b[0];
        return a[1] - b[1];
    })
    
    var answer = 0;
    var end = 0 ; 
    
    for(let i=0; i<targets.length; i++){
        if(targets[i][0] >= end){
            answer ++;
            end = targets[i][1];
        }
    }
    
    
    return answer;
}

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.java
import java.util.*;

class Solution {
    public int solution(int[][] targets) {
        
        Arrays.sort(targets, (o1, o2) -> {
            if(o1[1] == o2[1]) return o1[0] - o2[0];
            return o1[1] - o2[1];
        });  
        
        int answer = 0;
        int pos = 0;
        
        for(int i=0; i<targets.length; i++){
            if(targets[i][0] >= pos){
                answer ++;
                pos = targets[i][1];
            }
        }
        
        return answer;
    }
}